r1cs/core/src/lib.rs

```diff
@@ -64,6 +66,37 @@ pub enum Index {
     Aux(usize),
 }
 
+impl CanonicalSerialize for Index {
+    fn serialize<W: Write>(&self, mut writer: W) -> Result<(), SerializationError> {
+        let (is_input, size) = match self {
+            Index::Input(size) => (true, size),
+            Index::Aux(size) => (false, size),
+        };
+        CanonicalSerialize::serialize(&is_input, &mut writer)?;
+        writer.write_all(&(*size as u32).to_le_bytes())?;
+        Ok(())
+    }
+
+    fn serialized_size(&self) -> usize {
+        1 + std::mem::size_of::<u32>()
+    }
+}
+
+impl CanonicalDeserialize for Index {
```

Comment:
Is there canonical_serialize_deserialize_test somewhere that also test that the Index deserialization works correctly given its serialization implementation ?

Reply:
Added unit test for Index serialization and deserialization